internal/authentication/azure/auth.go

```diff
@@ -130,40 +97,50 @@ func (s EnvironmentSettings) GetTokenCredential() (azcore.TokenCredential, error
 		}
 	}
 
-	// 3. MSI with timeout of 1 second (same as DefaultAzureCredential)
-	var msiCred *azcore.TokenCredential
-	{
+	if tokencred, err := azidentity.NewChainedTokenCredential(creds, nil); err == nil {
+		return tokencred, nil
+	}
+
+	// 3. MSI + Azure CLI Credential in parallel. First one to succeed wins.
+
+	wg := sync.WaitGroup{}
+	ch := make(chan azcore.TokenCredential, 1)
+	wg.Add(2)
+
+	go func(wg *sync.WaitGroup, ch chan<- azcore.TokenCredential) {
+		defer wg.Done()
 		c := s.GetMSI()
-		msiCred, err := c.GetTokenCredential()
+		cred, err := c.GetTokenCredential()
 		if err == nil {
-			creds = append(creds, &timeoutWrapper{cred: msiCred, authmethod: "managed identity", timeout: 1 * time.Second})
+			ch <- cred
 		} else {
 			errs = append(errs, err)
 		}
-	}
+	}(&wg, ch)
 
-	// 4. AzureCLICredential
-	{
+	go func(wg *sync.WaitGroup, ch chan<- azcore.TokenCredential) {
+		defer wg.Done()
+		// sleep for 1 second to allow MSI to win before trying Azure CLI
+		time.Sleep(1 * time.Second)
 		cred, credErr := azidentity.NewAzureCLICredential(nil)
 		if credErr == nil {
-			creds = append(creds, &timeoutWrapper{cred: cred, authmethod: "Azure CLI", timeout: 5 * time.Second})
+			ch <- cred
 		} else {
 			errs = append(errs, credErr)
```

Comment:
This is not concurrency-safe when running in a goroutine. If both goroutines try to append an error at the same time, it will panic.

Instead, consider having a separate channel that is used to send errors